great_expectations/data_context/store/inline_store_backend.py

```diff
@@ -55,42 +66,53 @@ def __init__(
     def config(self) -> dict:
         return self._config
 
-    def _get(self, key: Tuple[str, str]) -> Any:
-        config_var_type: str = key[0]
-        config_var_name: str = key[1]
+    def _get(self, key: Tuple[str, ...]) -> Any:
+        (
+            resource_type,
+            resource_name,
+        ) = InlineStoreBackend._determine_resource_type_and_name_from_key(key)
+
+        project_config: DataContextConfig = self._data_context.config
+
+        if resource_type is DataContextVariableSchema.DATA_CONTEXT:
+            return project_config
 
-        variable_config: Any = self._data_context.config[config_var_type]
+        variable_config: Any = project_config[resource_type]
+
+        if resource_name is not None:
+            return variable_config[resource_name]
 
-        if config_var_name:
-            return variable_config[config_var_name]
         return variable_config
 
-    def _set(self, key: Tuple[str, str], value: Any, **kwargs: dict) -> None:
-        config_var_type: str = key[0]
-        config_var_name: str = key[1]
+    def _set(self, key: Tuple[str, ...], value: Any, **kwargs: dict) -> None:
+        (
+            resource_type,
+            resource_name,
+        ) = InlineStoreBackend._determine_resource_type_and_name_from_key(key)
 
         project_config: DataContextConfig = self._data_context.config
 
-        if config_var_name:
-            project_config[config_var_type][config_var_name] = value
+        if resource_type is DataContextVariableSchema.DATA_CONTEXT:
+            config_commented_map_from_yaml = yaml.load(value)
+            value = DataContextConfig.from_commented_map(
+                commented_map=config_commented_map_from_yaml
+            )
+            project_config = value
+        elif resource_name is not None:
+            project_config[resource_type][resource_name] = value
         else:
-            project_config[config_var_type] = value
+            project_config[resource_type] = value
 
-        try:
-            self._data_context._save_project_config()
-        except AttributeError as e:
-            logger.warning(
-                f"DataContext of type {type(self._data_context)} used with {self.__class__.__name__} is not file-system enabled: {e}"
-            )
+        self._persist_changes()
 
     def _move(
-        self, source_key: Tuple[str, str], dest_key: Tuple[str, str], **kwargs: dict
+        self, source_key: Tuple[str, ...], dest_key: Tuple[str, ...], **kwargs: dict
```

Comment:
Length of the tuple is variable due to the possibility of different keys and store "frontends"